RELEASE/scripts/autoscend/paths/wereprofessor.ash

```diff
@@ -0,0 +1,272 @@
+boolean in_wereprof()
+{
+	return my_path() == $path[WereProfessor];
+}
+
+void wereprof_initializeSettings()
+{
+	if(!in_wereprof())
+	{
+		return;
+	}
+	set_property("auto_wandOfNagamar", false);		//wand not used in this path
+	set_property("auto_wereprof_init", false); 		//string used for when we find smashed equipment so we know we don't need to look there
+	cli_execute('wereprofessor research');			//parse the research bench
+}
+
+boolean is_werewolf()
+{
+	if(!in_wereprof())
+	{
+		return false;
+	}
+	if(have_effect($effect[Savage Beast]) > 0)
+	{
+		return true;
+	}
+	return false;
+}
+
+void wereprof_buySkills()
+{
+	if(!in_wereprof())
+	{
+		return;
+	}
+	int rp = get_property("wereProfessorResearchPoints").to_int();
+	if(is_werewolf() || rp < 10)
+	{
+		return; // can't access the research bench as a werewolf and don't care about it when we have less than 10 RP
+	}
+	boolean do_skills = true;
+	if(get_property("wereProfessorTransformTurns").to_int() > 3)
+	{
+		do_skills = false; //Want as many RP as possible before looping through the skills
+	}
+	if((!is_werewolf() && turns_played() == 0))
+	{
+		auto_log_info("Buy skills first", "blue");
+		do_skills = true; //Do skills before we do anything else
+	}
+	if(organsFull() && my_adventures() <= auto_advToReserve() && !(contains_text(get_property("beastSkillsKnown").to_string(), "stomach3") && contains_text(get_property("beastSkillsKnown").to_string(), "liver3")))
+	{
+		auto_log_info("Need more organs", "blue");
+		do_skills = true; //If organs are full, should do skills if we need more organ space and don't have all organ expanding skills and limited adventures left
+	}
+	int cantbuy;
```

Comment:
add check for `beastSkillsAvailable` being empty. If it is, visit the `place.php?whichplace=wereprof_cottage&action=wereprof_researchbench`

Pref was empty and when I attempted to buy skills, got an infinite loop. When I manually visited the research bench, pref got updated. Ran autoscend again and no issues, bought skills as expected

Reply:
Yes, that is why I added cli_execute(wereprofessor research); to wereprof_initializeSettings. I will add it to the skill buying part too